### PR TITLE
Adding requirements

### PR DESCRIPTION
**Pull request by rokas-migauskas**
Threads nepadariau, also no clue ar gerai cia isvis prideta

### CODE REVIEW

File: aiKart/Exceptions/EntityValidationException.cs
Comment (MartynasJonkus) [inline aiKart/Exceptions/EntityValidationException.cs:1]: we need a generic method/event/delegate, not a class

**Review by MartynasJonkus — CHANGES_REQUESTED** [review]
Async/await usage is questionable and no generic method/event/delegate
Exception handling is probably good, but didn't examine it thoroughly

File: aiKart/Services/DeckService.cs
[the remark this answers is not in the document]
Comment (rokas-migauskas) [reply aiKart/Services/DeckService.cs:1]: deck service naudojau kaip task

File: aiKart/Repositories/DeckRepository.cs
[the remark this answers is not in the document]
Comment (rokas-migauskas) [reply aiKart/Repositories/DeckRepository.cs:1]: I think that using await through the whole call stack of a method is best practice, also, concerning the no code following part, the method called afterwards is asynchronous, and even if it wasn't, this part of the program uses middleware, which could mess with stuff if there is no await.

File: aiKart/Services/DeckService.cs
[the remark this answers is not in the document]
Comment (MartynasJonkus) [reply aiKart/Services/DeckService.cs:1]: Zinau, bet tas task niekur nepanaudotas

[the remark this answers is not in the document]
Comment (rokas-migauskas) [reply aiKart/Services/DeckService.cs:1]: ai, galiu pridet greit tada

**Review by MartynasJonkus — APPROVED** [review]
okay

**Review by K-Karalius — APPROVED** [review]

**Pull request merged**